mirage/factories/vendor.js

```diff
@@ -1,29 +1,33 @@
-import { Factory, faker } from 'mirage-server';
+import { Factory, faker, trait } from 'mirage-server';
 
 export default Factory.extend({
   vendorName: () => faker.company.companyName(),
   packagesTotal: 0,
   packagesSelected: 0,
 
-  afterCreate(vendor, server) {
-    // If packagesTotal is greater than zero, we'll auto-create the packages
-    // for this vendor here.
+  withPackagesAndTitles: trait({
+    afterCreate(vendor, server) {
+      // If packagesTotal is greater than zero, we'll auto-create the packages
+      // for this vendor here.
 
-    if(vendor.packagesTotal > 0) {
-      // Decide how many will be selected (0-packagesTotal)
-      vendor.packagesSelected = Math.floor(Math.random() * (vendor.packagesTotal + 1));
+      if(vendor.packagesTotal > 0) {
+        // Decide how many will be selected (0 to packagesTotal)
+        vendor.packagesSelected = Math.floor(Math.random() * (vendor.packagesTotal + 1));
 
-      server.createList('package', vendor.packagesSelected, {
-        vendor,
-        vendorName: vendor.vendorName,
-        isSelected: 1
-      });
+        server.createList('package', vendor.packagesSelected, 'withTitles', {
+          vendor,
+          vendorName: vendor.vendorName,
+          isSelected: true,
+          titleCount: Math.ceil(Math.random() * 5)
```

Comment:
very minor nit, but faker has a random number method that's a lot more readable: https://github.com/Marak/faker.js/wiki/Basic-Random-Data